### PR TITLE
Add-favourites

### PR DESCRIPTION
**Pull request by UtsavK55**

### CODE REVIEW

File: src/components/favouriteMovieCard/FavouriteMovieCard.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+import {styles} from './styles';
+
+const FavouriteMovieCard = ({movie}: {movie: MovieCardData}) => {
+  const navigation = useNavigation<MovieNavigationType>();
+  const {id, title, poster_path, vote_average, vote_count, overview} = movie;
+
+  return (
+    <TouchableOpacity
+      onPress={() =>
```
Comment (adiTrivsme) [inline src/components/favouriteMovieCard/FavouriteMovieCard.tsx:18]: define a function for this

File: src/components/favouriteMovieCard/FavouriteMovieCard.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+          <View style={styles.ratingContainer}>
+            <Rating vote_average={vote_average} />
+            <Text style={styles.info}>({vote_count})</Text>
+          </View>
+          <Text style={styles.info} numberOfLines={3}>
+            {overview}
+          </Text>
+        </View>
+        <TouchableOpacity
+          onPress={() =>
```
Comment (adiTrivsme) [inline src/components/favouriteMovieCard/FavouriteMovieCard.tsx:41]: extract to a function

File: src/components/favouriteMovieCard/styles.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import {colors} from '@theme/themes';
+import {StyleSheet} from 'react-native';
```
Comment (adiTrivsme) [inline src/components/favouriteMovieCard/styles.ts:2]: fix order of imports

File: src/components/favouriteMovieCard/styles.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+export const styles = StyleSheet.create({
+
```
Comment (adiTrivsme) [inline src/components/favouriteMovieCard/styles.ts:5]: no need for blank line here

File: src/components/favouriteMovieCard/styles.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    borderRadius: 8,
+    marginRight: 12,
+  },
+  movieInfo: {
+    flex: 1,
+    gap: 10,
+    marginRight: 10,
+  },
+  movieTitle: {
+    fontSize: 22,
```
Comment (adiTrivsme) [inline src/components/favouriteMovieCard/styles.ts:31]: use constants for font-related properties - sizes, family, etc

File: src/screens/favouritesScreen/FavouritesScreen.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import NoDataFound from '@components/noDataFound';
+
+import {accountId} from '@constants/token';
+import usePagination from '@hooks/usePagination';
+import {account} from '@network/apiUrls';
+
+import {colors} from '@theme/themes';
+
+const FavouritesScreen = () => {
+
```
Comment (adiTrivsme) [inline src/screens/favouritesScreen/FavouritesScreen.tsx]: no need for blank line at start

File: src/screens/favouritesScreen/FavouritesScreen.tsx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+          onEndReached={loadMore}
+          onEndReachedThreshold={0.1}
+          ListEmptyComponent={<NoDataFound item="favourite movies" />}
+        />
+      </View>
+    </BaseContainer>
+  );
+};
+
+const styles = StyleSheet.create({
```
Comment (adiTrivsme) [inline src/screens/favouritesScreen/FavouritesScreen.tsx]: put styles in a separate screen

File: src/components/favouriteMovieCard/FavouriteMovieCard.tsx [hunk 8]
```diff
@@ -0,0 +1,56 @@
+import {Image, Text, TouchableOpacity, View} from 'react-native';
+import {useNavigation} from '@react-navigation/native';
+import Icon from 'react-native-vector-icons/Feather';
+
+import Rating from '@components/rating';
+import {colors} from '@theme/themes';
+import {ROUTES} from '@constants/Routes';
+import {imageUrl} from '@helpers/helper';
```
[the remark this answers is not in the document]
Comment (UtsavK55) [reply src/components/favouriteMovieCard/FavouriteMovieCard.tsx:8]: imageUrl is a function.
export const imageUrl = (backdrop_path: string) => {
  return `https://image.tmdb.org/t/p/w500${backdrop_path}`;
};

File: src/components/favouriteMovieCard/styles.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+      borderRadius: 8,
+      marginRight: 12,
+    },
+    movieInfo: {
+      flex: 1,
+      gap: 10,
+      marginRight: 10,
+    },
+    movieTitle: {
+      fontSize: spacing.l,
```
Comment (adiTrivsme) [inline src/components/favouriteMovieCard/styles.ts:33]: define constants for fonts, this is not clear and also pose the risk of messing up constants when updating some spacing issues

**Review by adiTrivsme — APPROVED** [review]

**Pull request merged**